src/main/java/io/github/nucleuspowered/nucleus/commands/admin/ClearInventoryCommand.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * This file is part of Nucleus, licensed under the MIT License (MIT). See the LICENSE.txt file
+ * at the root of this project for more details.
+ */
+package io.github.nucleuspowered.nucleus.commands.admin;
+
+import io.github.nucleuspowered.nucleus.Util;
+import io.github.nucleuspowered.nucleus.api.PluginModule;
+import io.github.nucleuspowered.nucleus.internal.CommandBase;
+import io.github.nucleuspowered.nucleus.internal.annotations.Modules;
+import io.github.nucleuspowered.nucleus.internal.annotations.NoCooldown;
+import io.github.nucleuspowered.nucleus.internal.annotations.NoCost;
+import io.github.nucleuspowered.nucleus.internal.annotations.NoWarmup;
+import io.github.nucleuspowered.nucleus.internal.annotations.RegisterCommand;
+import org.spongepowered.api.command.CommandResult;
+import org.spongepowered.api.command.CommandSource;
+import org.spongepowered.api.command.args.CommandContext;
+import org.spongepowered.api.command.args.GenericArguments;
+import org.spongepowered.api.command.spec.CommandSpec;
+import org.spongepowered.api.entity.living.player.Player;
+import org.spongepowered.api.text.Text;
+
+import java.util.Optional;
+
+@RegisterCommand({"clear", "clearinv", "clearinventory"})
+@Modules(PluginModule.ADMIN)
+@NoCooldown
+@NoWarmup
+@NoCost
+public class ClearInventoryCommand extends CommandBase<CommandSource> {
+
+    private final String player = "player";
+
+    @Override
+    public CommandSpec createSpec() {
+        return CommandSpec.builder().arguments(GenericArguments.onlyOne(GenericArguments.optional(GenericArguments.player(Text.of(player)))))
+                .executor(this).build();
+    }
+
+    @Override
+    public CommandResult executeCommand(CommandSource src, CommandContext args) throws Exception {
+        Optional<Player> pl = args.<Player>getOne(player);
```

Comment:
https://github.com/NucleusPowered/Nucleus/blob/master/src/main/java/io/github/nucleuspowered/nucleus/commands/misc/HealCommand.java#L50-L53

This is how I handle commands like this where I have an optional user for console usage. It includes this check, and sends a message. That way, you know you have a player target, and checking for whether you are acting on self or another player is simply an equality check.